rotkehlchen/premium/sync.py

```diff
@@ -31,11 +34,11 @@ class SyncCheckResult(NamedTuple):
 
 class PremiumSyncManager():
 
-    def __init__(self, data: DataHandler, password: str) -> None:
+    def __init__(self, rotki: 'Rotkehlchen', password: str) -> None:
         # Initialize this with the value saved in the DB
-        with data.db.conn.read_ctx() as cursor:
-            self.last_data_upload_ts = data.db.get_setting(cursor, name='last_data_upload_ts')
-        self.data = data
+        with rotki.data.db.conn.read_ctx() as cursor:
+            self.last_data_upload_ts = rotki.data.db.get_setting(cursor, name='last_data_upload_ts')  # noqa: E501
+        self.rotki = rotki
```

Comment:
no no no. You can't give the super object as an attribute here. This is very very bad design. Just pass the data migration manager alone. And the data handler as it was before.